NLog.Windows.Forms/RichTextBoxTarget.cs

```diff
@@ -215,52 +242,66 @@ protected override void InitializeTarget()
-                FormName = "NLogForm" + Guid.NewGuid().ToString("N");
+                FormName = "NLogForm" + Guid.NewGuid().ToString("N");
+                InitWithNewForm();
+                return;
+            }
+
+            if (string.IsNullOrEmpty(ControlName))
+            {
+                string message = "Rich text box control name must be specified for " + GetType().Name + ".";
+                if (LogManager.ThrowExceptions)
+                {
+                    throw new NLogConfigurationException(message);
+                }
+                InternalLogger.Error(message);
+                InitWithNewForm();
+                return;
             }
-            else
-            {
-                var openFormByName = Application.OpenForms[FormName];
-                if (openFormByName != null)
-                {
-                    bool error = false;
-                    TargetForm = openFormByName;
-                    if (string.IsNullOrEmpty(ControlName))
-                    {
-                        error = true;
-                        var message = "Rich text box control name must be specified for " + GetType().Name + ".";
-                        if (LogManager.ThrowExceptions)
-                        {
-                            throw new NLogConfigurationException(message);
-                        }
-
-                        InternalLogger.Error(message);
-                    }
 
-                    CreatedForm = false;
-                    TargetRichTextBox = FormHelper.FindControl<RichTextBox>(ControlName, TargetForm);
+            Form openFormByName = Application.OpenForms[FormName];
+            if (openFormByName == null)
+            {
+                if (AllowCustomFormCreation)
+                {
+                    InternalLogger.Info("Form {0} not found, creating a new one.", FormName);
+                    InitWithNewForm();
+                }
+                else
+                {
+                    InternalLogger.Info("Form {0} not found, waiting for ReInitializeAllTextboxes.", FormName);
+                }
+                return;
+            }
 
-                    if (TargetRichTextBox == null)
-                    {
-                        error = true;
-                        var message = "Rich text box control '" + ControlName + "' cannot be found on form '" + FormName + "'.";
-                        if (LogManager.ThrowExceptions)
-                        {
-                            throw new NLogConfigurationException(message);
-                        }
-                        InternalLogger.Error(message);
-                    }
-                    if (!error)
-                        return;
+            TargetForm = openFormByName;
+            CreatedForm = false;
+            TargetRichTextBox = FormHelper.FindControl<RichTextBox>(ControlName, TargetForm);
+            if (TargetRichTextBox == null)
```

Comment:
Not sure what duplication you meant here, but I've extracted repeated code into `DoException` method.